test/app-luatest/gh_7479_bug_fix_uri_lib_test.lua

```diff
@@ -0,0 +1,90 @@
+local t = require('luatest')
+local g = t.group()
+local net_box = require('net.box')
+
+-- Checks whether it is possible to specify only port for localhost connection.
+g.test_tarantoolctl_connect_with_only_port_indication = function()
+    local tarantoolctl_location = io.popen('which tarantoolctl'):read()
+    -- Code variable below contains logic of test instance.
+    -- Also it contains logic of tarantoolctl connect launch to test instance.
+    local remote_code = [==[
+        local cmd_test_instance = {
+            arg[-1],
+            '-e',
+            [[
+                box.cfg{listen = 0}
+                box.schema.user.grant('guest', 'read, write, execute', 'universe')
+            ]],
+        }
+
+        -- Test instance will be started in a separate process.
+        local ph_test_instance = require('popen').new(cmd_test_instance, {
+            stdin  = 'devnull',
+            stdout = 'devnull',
+            stderr = require('popen').opts.PIPE,
+        })
+        os.execute("sleep " .. tonumber(1))
+
+        -- In /tmp/port-gh-7479.txt we will save auto-assigned port.
+        local file, err = io.open('/tmp/port-gh-7479.txt', 'w')
+        if file == nil then
+            error(err)
+        end
+        -- Get test instance init message that contains port information.
+        local test_instance_init_msg = ph_test_instance:read({stderr = true}):rstrip()
+        -- Get substring like *bound to [::]:49803* in instance init message.
+        local port = string.match(test_instance_init_msg, "[::]:]:%s*(%S+)")
+        if port == nil then
+            -- Get substring like *bound to 0.0.0.0:49803* in instance init message.
+            port = string.match(test_instance_init_msg, "0.0.0.0:%s*(%S+)")
+        end
+        file:write(port)
+        os.execute("sleep " .. tonumber(1))
+
+        local cmd_tarantoolctl = {
+            ']==] .. tarantoolctl_location .. [==[',
+            'connect',
+            -- Indicate only port, without localhost:* before the port value.
+            port,
+        }
+        -- Connection will be established in a separate process.
+        local ph_tarantoolctl = require('popen').new(cmd_tarantoolctl, {
+            stdin  = 'devnull',
+            stdout = 'devnull',
+        })
+        os.execute("sleep " .. tonumber(1))
+        ph_test_instance:close()
+        ph_tarantoolctl:close()
+    ]==]
```

Comment:
Better put the source in a separate file instead of writing it to a file from a string on the fly. Anyway, I don't understand why you need a separate source file. Can't you create a [server](https://github.com/tarantool/tarantool/blob/master/test/luatest_helpers/server.lua) and start a console in it?

```lua
require('console').listen(0)
```

BTW this function returns the port, the console listens to.

Reply:
Done.  Thanks for pointing to `console.listen(0)`, it really simplified the solution